src/history/hs_rec.c

```diff
@@ -672,7 +672,18 @@ __wt_hs_insert_updates(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_MULTI *mult
                 F_SET(upd, WT_UPDATE_HS);
                 if (tombstone != NULL)
                     F_SET(tombstone, WT_UPDATE_HS);
+                if (ret == 0) {
```

Comment:
So we only flag updates if the cursor states the insertion was successful and if the error code returned was 0, I do this because I'm unsure if EBUSY can get returned here due to OOO updates.

Reply:
Have we agreed to only flag it when all the updates are inserted to the history store?

Reply:
We have agreed to flag it after all HS inserts are done for that key. Resolving this conversation as the code has since moved.